### ISSUE TITLE
[Research][Struct2depth]Do rectangular masks for object motion work?

### ISSUE THREAD

**Issue by poornimajd** [tensorflow/models#8430]
Hi great work! @aneliaangelova  and team.
I am trying to generate masks for training the model to even learn object motion.
I have bounding boxes for my dataset,but to generate masks from it ,is it necessary for  it to be Run length encoding (RLE) format (used by Mask-RCNN) ?
I was wondering if it is fine to just generate masks from bounding box by 
[xmin,ymin:xmax,ymax]=[255,255,255] , and another instance [254,254,254] and so on ,so basically I will lose the shape of the object and all masks will be rectangular in shape.Since the model does not learn the shape but just the motion , so is this method correct or does it require the RLE masks itself?
Any suggestion is greatly appreciated.
Thank you

**Comment by aneliaangelova** [comment]
Hi, 
Your question has two aspects:
-We don't use RLE at all, we only need MaskRCNN to give us the mask. A mask of an object is just pixel-based mask. For example for object number 1 in the image we'll assign 1 in the mask to all pixels that belong to this object. The N-th object in the same image will be assigned the id N in the mask to all pixels that belong to it.  So the mask should be moslty 0s and numbers which correspond to ids of object instances.
-While we used masks, you can use the bounding boxes as masks and relax this condition. Please take a look at out subsequent work which needs a much cruder mask:
Paper "Depth from Videos in the Wild": http://openaccess.thecvf.com/content_ICCV_2019/papers/Gordon_Depth_From_Videos_in_the_Wild_Unsupervised_Monocular_Depth_Learning_ICCV_2019_paper.pdf
Code: https://github.com/google-research/google-research/tree/master/depth_from_video_in_the_wild
Thanks!

**Comment by poornimajd** [comment]
Thanks a lot for the quick reply and the detailed explanation! I also checked out the paper in brief and it looks good to use bounding box as masks.
Also I just had another question.
I do not have the extrinsic parameters for camera to lidar to evaluate the trained model for depth or motion on the Indian Driving Dataset,is there any other work around or should I use something like the loss used while training the network?
I was trying to use https://github.com/tinghuiz/SfMLearner for the evaluation part of the model.
Any suggestion is greatly appreciated .
Thank you

**Comment by aneliaangelova** [comment]
You can use standard datasets KITTI and Cityscapes which have the depth ground truth for evaluation( also the ETH MAV data one used in the paper above). Other public self-driving datasets do have the proper extrinsics and stereo can also be used as a proxy for gt depth.

**Comment by poornimajd** [comment]
Thanks for the reply! I have trained my model on Indian driving dataset(IDD),which is more complex than kitti and cityscapes,so I am not sure if I can use their ground truth to check the accuracy of model trained on IDD.I do not have the extrinsics too for the IDD.I tried some stereo matching methods  for depth from stereo but did not get great results.
I am not sure if this is what was suggested by you,any corrections very much appreciated.
Thank you and sorry for the trouble.
